src/driver/mod.rs

```diff
@@ -145,10 +145,39 @@ impl Drop for Driver {
 
         // pre-determine what to cancel
         let mut cancellable_ops = Vec::new();
-        for (id, cycle) in self.ops.lifecycle.iter() {
-            // don't cancel completed items
-            if !matches!(cycle, Lifecycle::Completed(_)) {
-                cancellable_ops.push(id);
+
+        let Ops {
+            lifecycle,
+            completions,
+        } = &mut self.ops;
+
+        for (id, cycle) in lifecycle.iter_mut() {
+            match cycle {
+                Lifecycle::Completed(_) => {
+                    // don't cancel completed items
+                }
+
+                Lifecycle::CompletionList(indices) => {
+                    let mut list = indices.clone().into_list(completions);
+                    if io_uring::cqueue::more(list.peek_end().unwrap().flags) {
+                        // If the completion list is not finished,
+                        // mark for cancellation, and mark the lifecycle as ignored
+                        cancellable_ops.push(id);
+                        *cycle = Lifecycle::Ignored(Box::new(()));
+                        // Dropping the list here deallocates the unconsumed cqe slab entries
+                    } else {
+                        // This op is complete. Replace with a null Completed entry
```

Comment:
Interesting the last peeked entry, found not to be More (so it represents finished), could be put into the Completed, result and flags fields, but I also don't see the point. But saving Ok(0) seems odd. I don't have a better alternative though.